chore: implement use-using rule

jrandolf-google opened the pull request:
This is an internal TypeScript plugin that enforces usage of `using` in our codebase.

OrKoN commented on tools/eslint/src/use-using.ts:
```diff
@@ -0,0 +1,73 @@
+import {ESLintUtils, TSESTree} from '@typescript-eslint/utils';
```
missing license headers

OrKoN replied:
here and other files

OrKoN reviewed (commented):
Otherwise, LGTM

jrandolf committed:
chore: implement use-using rule

This is an internal eslint rule that enforces usage of `using` in our codebase.
